exchanges/graphcache/src/cacheExchange.test.ts

```diff
@@ -1448,15 +1609,17 @@ describe('optimistic updates', () => {
 
     expect(response).toHaveBeenCalledTimes(1);
     expect(optimistic.concealAuthor).toHaveBeenCalledTimes(2);
-    expect(reexec).toHaveBeenCalledTimes(2);
+    expect(reexec).toHaveBeenCalledTimes(1);
```

Comment:
Have to verify these changes first though 😅

Reply:
The change in test behaviour seems as intended.

Above we have two mutations that both trigger the same optimistic change, hence, one doesn't actually change the value, so the swallowed change is correct.

However, I'm a little curious whether we're at risk of violating some layering if we don't analyse that first since there's a slight possibility of shifted layers (layers hosited up) not seeing an actual change, even if they have the side-effect of overriding a field back to an updated value.
i.e. from the `write`’s perspective nothing changes, but from the `read` perspective a value _does_ change.

Shouldn't be too much of a big deal though if that does happen (?)
But does highlight that, potentially, the layering can be simplified as we talked about before